third_party/terraform/website/docs/r/compute_router_nat.html.markdown

```diff
@@ -0,0 +1,121 @@
+---
+layout: "google"
+page_title: "Google: google_compute_router_nat"
+sidebar_current: "docs-google-compute-router-nat"
+description: |-
+  Manages a Cloud NAT.
+---
+
+# google\_compute\_router\_nat
+
+Manages a Cloud NAT. For more information see
+[the official documentation](https://cloud.google.com/nat/docs/overview)
+and
+[API](https://cloud.google.com/compute/docs/reference/rest/beta/routers).
+
+## Example Usage
+
+A simple NAT configuration: enable NAT for all Subnetworks associated with
+the Network associated with the given Router.
+
+```hcl
+resource "google_compute_router_nat" "simple-nat" {
+  name                               = "nat-1"
+  router                             = "router-1"
+  region                             = "us-central1"
+  nat_ip_allocate_option             = "AUTO_ONLY"
+  source_subnetwork_ip_ranges_to_nat = "ALL_SUBNETWORKS_ALL_IP_RANGES"
+}
+```
+
+A production-like configuration: enable NAT for one Subnetwork and use a list of
+static external IP address.
+
+```hcl
+resource "google_compute_router_nat" "advanced-nat" {
+  name                               = "nat-1"
+  router                             = "router-1"
+  region                             = "us-central1"
+  nat_ip_allocate_option             = "MANUAL_ONLY"
+  nat_ips                            =
```

Comment:
can you also add resource definitions for anything referenced in this example? Also, consider whether it makes sense to use `count = 2` on the address resource so you can use the splat operator to get all the addresses without having to specify each individually

Reply:
Done.